internal/dangling/dangling.go

```diff
@@ -117,6 +119,21 @@ func DeleteUnusedMachineConfigs(cli client.Client, ctx context.Context, instance
 	return errors
 }
 
+func keepMachineConfig(expected bool, mc machineconfigv1.MachineConfig, filterObjects ...KeepDanglingObjectFunc) bool {
+	if len(filterObjects) == 0 {
+		return expected
+	}
+	for _, filterObject := range filterObjects {
+		keep, reason := filterObject(expected, &mc)
+		if keep {
```

Comment:
I don't see any good reason for keeping a dangling object.
Moreover we should expose a way for the user to specify whether he wants that or not.